stock_split_picking/models/stock_picking.py

```diff
@@ -0,0 +1,111 @@
+                raise UserError(
+                    _(
+                        "You must enter done quantity in order to split your "
+                        "picking in several ones."
+                    )
+                )
+
+            # Split moves considering the qty_done on moves
+            new_moves = self.env["stock.move"]
+            for move in picking.move_ids:
+                rounding = move.product_uom.rounding
+                qty_done = move.quantity_done
+                qty_initial = move.product_uom_qty
+                qty_diff_compare = float_compare(
+                    qty_done, qty_initial, precision_rounding=rounding
+                )
+                if qty_diff_compare < 0:
+                    qty_split = qty_initial - qty_done
+                    qty_uom_split = move.product_uom._compute_quantity(
+                        qty_split, move.product_id.uom_id, rounding_method="HALF-UP"
+                    )
+                    new_move_vals = move._split(qty_uom_split)
+                    for move_line in move.move_line_ids:
+                        if move_line.reserved_qty and move_line.qty_done:
+                            # To avoid an error
+                            # when picking is partially available
+                            try:
+                                move_line.write(
+                                    {"reserved_uom_qty": move_line.qty_done}
+                                )
+                            except UserError:
+                                continue
+                    new_move = self.env["stock.move"].create(new_move_vals)
+                    new_move._action_confirm(merge=False)
+                    new_moves |= new_move
+
+            # If we have new moves to move, create the backorder picking
+            if new_moves:
+                backorder_picking = picking._create_split_backorder()
+                new_moves.write({"picking_id": backorder_picking.id})
+                new_moves.mapped("move_line_ids").write(
+                    {"picking_id": backorder_picking.id}
+                )
+                new_moves._action_assign()
+
+    def _create_split_backorder(self, default=None):
+        """Copy current picking with defaults passed, post message about
+        backorder"""
+        self.ensure_one()
+        backorder_picking = self.copy(
+            dict(
+                {
+                    "name": "/",
+                    "move_ids": [],
+                    "move_line_ids": [],
+                    "backorder_id": self.id,
+                },
+                **(default or {})
+            )
+        )
+        self.message_post(
+            body=_(
+                'The backorder <a href="#" '
+                'data-oe-model="stock.picking" '
+                'data-oe-id="%(id)d">%(name)s</a> has been created.'
+            ),
+            id=backorder_picking.id,
+            name=backorder_picking.name,
+        )
```

Comment:
```suggestion
        self.message_post(
            body=_(
                'The backorder %s has been created.'
            , backorder_picking._get_html_link())
        )
```

Reply:
@ps-tubtim

Reply:
@AungKoKoLin1997 @rousseldenis  I already change it.